### PR TITLE
Installer adjustments for clang 13.1.6+, and some mac arm64 adjustments

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Installer adjustments for clang 13.1.6+ (any OS really)
Adjustments for arm64

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: ConanInstall.cmake [hunk 1]
```diff
@@ -60,7 +60,7 @@ if(NOT CONAN_OPENSTUDIO_ALREADY_RUN)
   list(APPEND CONAN_BUILD "missing")
 
   if (BUILD_TESTING)
-    set(CONAN_GTEST "gtest/1.10.0#ef88ba8e54f5ffad7d706062d0731a40")
+    set(CONAN_GTEST "gtest/1.10.0#2210002307b82e555f7bec6c9ec6c839")
```
Comment (jmarrec) [inline ConanInstall.cmake:63]: The other revision doesn't build on Apple clang version 13.1.6 which become more stringent with warnings

File: python/setup.py [hunk 2]
```diff
@@ -45,10 +45,13 @@ def get_tag(self):
...
-                # but we set the platform to old one in the hope that it'll
-                # work for all
-                plat = 'macosx_10_9_x86_64'
+                if 'arm64' in plat:
+                    plat = 'macosx_11_0_arm64'
+                else:
+                    # We don't use a fat binary ('intel' = both i386 and x86_64)
+                    # but we set the platform to old one in the hope that it'll
+                    # work for all
+                    plat = 'macosx_10_9_x86_64'
```
Comment (jmarrec) [inline python/setup.py:54]: python platform override for arm64.

File: src/energyplus/Test/ScheduleInterval_GTest.cpp [hunk 3]
```diff
@@ -1569,7 +1569,7 @@ TEST_F(EnergyPlusFixture, DISABLED_ForwardTranslator_ScheduleVariableInterval_Da
   bool until24Found = false;
   bool nextValueShouldBeLast = false;
   bool hitNextValueShouldBeLast = false;
-  unsigned numUntils = 0;
+  [[maybe_unused]] unsigned numUntils = 0;
```
Comment (jmarrec) [inline src/energyplus/Test/ScheduleInterval_GTest.cpp:1572]: fix some `-Wunused-but-set-variable` on apple clang 13.1.6 (earlier versions won't have the issue)

File: src/model/test/Space_GTest.cpp [hunk 4]
```diff
@@ -79,6 +79,44 @@
 using namespace openstudio;
 using namespace openstudio::model;
 
+struct SurfaceAreaCounter
```
Comment (jmarrec) [inline src/model/test/Space_GTest.cpp:82]: this is for -Wunused-but-set-variable on apple clang 13.1.6, but I also used the occasion to dry up the code

File: CMakeLists.txt [hunk 5]
```diff
@@ -1247,6 +1247,7 @@ set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${OPENSTUDIO_LONG_VERSION}-${
 set(CPACK_PACKAGE_CONTACT "openstudio@nrel.gov")
 
 if(APPLE)
+  set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${OPENSTUDIO_LONG_VERSION}-${CMAKE_SYSTEM_NAME}-${ARCH}")
```
[the remark this answers is not in the document]
Comment (jmarrec) [reply CMakeLists.txt:1250]: @tijcolem I'm not 100% sure what naming convention we want.

Darwin-x86_64 and Darwin-arm64 is what this'll do, like E+ does, but that means changing a couple of locations like the new OS-res full installer test which computes the installer link.

**Review by tijcolem — APPROVED** [review]

**Pull request merged**